examples/remarketing/add_remarketing_action.py

```diff
@@ -0,0 +1,143 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This example demonstrates usage of remarketing actions.
+
+A new remarketing action will be created for the specified customer, and its
+associated tag snippets will be retrieved.
+"""
+
+
+import argparse
+import sys
+import uuid
+
+import google.ads.google_ads.client
+
+
+def main(client, customer_id, page_size):
+    remarketing_action_resource_name = _add_remarketing_action(
+        client, customer_id)
+
+    print(f'Created remarketing action "{remarketing_action_resource_name}".')
+
+    queried_remarketing_action = _query_remarketing_action(
+        client, customer_id, remarketing_action_resource_name, page_size)
+
+    _print_remarketing_action_attributes(client, queried_remarketing_action)
+
+
+def _add_remarketing_action(client, customer_id):
+    remarketing_action_service = client.get_service(
+        'RemarketingActionService', version='v2')
+
+    # Create the operation.
+    remarketing_action_operation = client.get_type(
+        'RemarketingActionOperation', version='v2')
+
+    # Create remarketing action.
+    remarketing_action = remarketing_action_operation.create
+    remarketing_action.name.value = f'Remarketing action #{uuid.uuid4()}'
+
+    # Add the remarketing action.
+    try:
+        remarketing_action_response = (
+            remarketing_action_service.mutate_remarketing_actions(
+                customer_id, [remarketing_action_operation]))
+    except google.ads.google_ads.errors.GoogleAdsException as ex:
+        print(f'Request with ID "{ex.request_id}" failed with status '
+              f'"{ex.error.code().name}" and includes the following errors:')
+        for error in ex.failure.errors:
+            print(f'\tError with message "{error.message}".')
+            if error.location:
+                for field_path_element in error.location.field_path_elements:
+                    print(f'\t\tOn field: {field_path_element.field_name}')
+        sys.exit(1)
+
+    return remarketing_action_response.results[0].resource_name
+
+
+def _query_remarketing_action(client, customer_id, resource_name, page_size):
+    # Creates a query that retrieves the previously created remarketing action
+    # with its generated tag snippets.
+    query = ('SELECT remarketing_action.id, '
+             'remarketing_action.name, '
+             'remarketing_action.tag_snippets '
+             'FROM remarketing_action '
+             f'WHERE remarketing_action.resource_name = "{resource_name}"')
+
+    google_ads_service_client = client.get_service(
+        'GoogleAdsService', version='v2')
+
+    # Issues a search request by specifying page size.
+    results = google_ads_service_client.search(
+        customer_id, query=query, page_size=page_size)
+
+    try:
+        for row in results:
```

Comment:
why have a `for` loop at all if it is just one row? 

`return results[0].remarketing_action`

Reply:
Because it's not a list or list-like object, it's a `GRPCIterator`. I wasn't sure iterating for the one remarketing action or using `list(results)[0].remarketing_action` was preferable, so I went with the option that didn't involve instantiating a new object. In hindsight though, maybe it'd be easier to read if I did.